libcloud/compute/drivers/verizon.py

```diff
@@ -0,0 +1,890 @@
+        result = self.connection.request(url).object['Items']
+        ssh_href = None
+        for i in result:
+            if i['Default']:
+                ssh_href = i['href']
+        if ssh_href is None:
+            raise VerizonException('No default key found')
+        return ssh_href
+
+    def ex_poweron_node(self, node=None):
+        """
+        Power on a node
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: status
+        :rtype : ``str``
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        href = None
+        for i in node.extra['Actions']:
+            if i is not None:
+                if i['name'] == 'power:powerOn':
+                    href = i['href']
+        headers = {'Content-Length': 0}
+        return self.connection.request(href,headers=headers,method='POST').object
+
+    def ex_wait_create(self, node=None, timeout=900):
+        """
+        Wait for node to power on
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :keyword timeout: Time to wait until failing
+        :type    timeout: ``int``
+
+        :rtype : ``bool``
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        end_time = int(time.time()) + timeout
+        while True:
+            if self.ex_get_node_state(node) == 5:
+                break
+            if int(time.time()) > end_time:
+                raise VerizonException('Time %s exceeded' % timeout)
+            time.sleep(10)
+        return True
+
+    def ex_get_node_state(self, node=None):
+        """
+        Get node state
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: State of the node
+        :rtype: :class: `Node`
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        result = self.connection.request(node.id).object
+        return self._to_node(result).state
+
+    def ex_get_node(self, node=None):
+        """
+        Returns a node
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: State of the node
+        :rtype: :class: `Node`
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        result = self.connection.request(node.id).object
+        return self._to_node(result)
+
+    def ex_poweroff_node(self, node=None):
+        """
+        Power off a node
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: State of the node
+        :rtype: :class: `Node`
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        href = None
+        for i in node.extra['Actions']:
+            if i is not None:
```

Comment:
Would make it slightly easier to read if you combined into a single lint - `if is is not None and i['name'] == 'power:powerOff'`